src/server/controllers/UserTransaction.controller.ts

```diff
@@ -0,0 +1,175 @@
+    }
+
+    res.status(HttpResponseCodesEnum.CREATED).json(resolveInfinitePaginationResponse(response));
+  }
+
+  async deleteById(req: NextApiRequest, res: NextApiResponse): Promise<void> {
+    const query: IdDTO = req.query;
+    validateSchema<IdDTO>(idDtoSchema, query);
+
+    const response: null | UserTransaction = await prisma.userTransaction.delete({
+      where: {
+        id: Number(query.id),
+      },
+    });
+
+    if (!response) {
+      throw new NotFoundException(`User Transaction, with id ${query.id}, not found`);
+    }
+
+    res.status(HttpResponseCodesEnum.OK).json(resolveInfinitePaginationResponse(response));
+  }
+
+  async getByHash(req: NextApiRequest, res: NextApiResponse): Promise<void> {
+    const query: HashDTO = req.query;
+    validateSchema<HashDTO>(hashDtoSchema, query);
+
+    const response: null | UserTransaction = await prisma.userTransaction.findUnique({
+      where: {
+        contractHash: query.hash,
+      },
+    });
+
+    if (!response) {
+      throw new NotFoundException(`User, with hash ${query.hash}, not found`);
+    }
+
+    res.status(HttpResponseCodesEnum.OK).json(resolveInfinitePaginationResponse(response));
+  }
+
+  async getById(req: NextApiRequest, res: NextApiResponse): Promise<void> {
+    const query: IdDTO = req.query;
+    validateSchema<IdDTO>(idDtoSchema, query);
+
+    const response: null | UserTransaction = await prisma.userTransaction.findUnique({
+      where: {
+        id: Number(query.id),
+      },
+    });
+
+    if (!response) {
+      throw new NotFoundException(`User Transaction, with id ${query.id}, not found`);
+    }
+
+    res.status(HttpResponseCodesEnum.OK).json(resolveInfinitePaginationResponse(response));
+  }
+
+  async getMany(req: NextApiRequest, res: NextApiResponse): Promise<void> {
+    const query: GetManyUserTransactionDTO = req.query as unknown as GetManyUserTransactionDTO;
+    const pagination: InfinitePaginationType = resolveInfinitePagination(req.query);
+    validateSchema<GetManyUserTransactionDTO>(getManyUserTransactionDtoSchema, query);
+
+    const response = await prisma.userTransaction.findMany({
+      where: resolveBulkArgs<Prisma.UserTransactionWhereInput>([
+        {
+          key: 'id',
+          value: query?.ids,
+        },
+        {
+          key: 'amount',
+          value: query?.amount,
+        },
+        {
+          key: 'contractHash',
+          value: query?.contractHashes,
+        },
+        {
+          key: 'userId',
+          value: query?.userId,
+        },
+      ]),
+      ...resolvePrismaPaginationArgs(pagination),
+    });
+
+    res
+      .status(HttpResponseCodesEnum.OK)
+      .json(resolveInfinitePaginationResponse(response, pagination));
+  }
+
+  async getTotalAmountByUser(req: NextApiRequest, res: NextApiResponse): Promise<void> {
+    try {
+      const allTransactions = await prisma.userTransaction.findMany({
```

Comment:
Lets remove the prisma...findMany from the try/catch block. If it throws an error, it will be handled by the global error handling